pkgs/development/compilers/ghc/8.10.4.nix

```diff
@@ -91,7 +91,7 @@ let
 
   # ld.gold is disabled for musl libc due to https://sourceware.org/bugzilla/show_bug.cgi?id=23856
   # see #84670 and #49071 for more background.
-  useLdGold = targetPlatform.isLinux && !(targetPlatform.useLLVM or false) && !targetPlatform.isMusl;
+  useLdGold = targetPlatform.isLinux && targetPlatform.linker == "bfd" && !targetPlatform.isMusl;
```

Comment:
What does `targetPlatform.linker == "bfd"` mean?

(Also, why does this mean that gold should be used?)

Reply:
`ld.bfd` is the classic GNU ld that uses the bfd library which was (at least historically) behind most of binutils and gdb. `ld.gold` is the alternative one that is a bit odd that it is even part of the binutils project at all. Had it been started a few years later, I assume it would have been an LLVM instead (and been more like LLD).

Reply:
https://www.phoronix.com/scan.php?page=news_item&px=GNU-Gold-Stagnate-F31

Reply:
Interestingly, if we moved it to a separate package too, it would probably be easier to implement `linker == "gold"`.

Reply:
> One other question I had: what made you want to change/fix this? Were you seeing a problem that targetPlatform.useLLVM wasn't catching?

`useLLVM` implies `linker == "lld"`, but we recently made it (theoretically) possible to have something like `useLLVM` without `linker == "lld"` (which was when `linker` was introduced in the first place). Thus checking `linker` is more accurate and future proof.